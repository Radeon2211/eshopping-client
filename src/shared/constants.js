import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
