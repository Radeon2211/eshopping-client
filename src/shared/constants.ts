import * as Yup from 'yup';
import countryCodes from 'country-codes-list';
import { ProductCondition, ProductPhotoFieldValue } from './types/enums';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([ProductCondition.NEW, ProductCondition.USED, ProductCondition.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
  photo: Yup.mixed()
    .test('is-error', (value) => value !== ProductPhotoFieldValue.ERROR)
    .nullable(),
};

export const listOfAreaCodes = countryCodes.customArray({
  value: '{countryCallingCode}',
  label: '+{countryCallingCode} {countryNameEn}',
});

export const listOfCountries = countryCodes.customArray({
  value: '{countryNameEn}',
  label: '{countryNameEn}',
});

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: 'Price - ascending' },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: 'Price - ascending' },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig: ScrollToOptions = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};
