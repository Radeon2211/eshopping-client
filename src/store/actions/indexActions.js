export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions';

export { setModal, deleteMessage, changeProductsPerPage } from './uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions';
