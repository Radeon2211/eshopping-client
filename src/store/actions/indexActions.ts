export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeEmail,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions/authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions/tradeActions';

export { setMessage, setModal, changeProductsPerPage } from './uiActions/uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions/productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions/orderActions';
