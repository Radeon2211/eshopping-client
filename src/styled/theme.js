const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
