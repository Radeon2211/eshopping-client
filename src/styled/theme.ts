const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#777',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#06AD96',
    greenDark: '#009680',
    red: '#F22613',
    redLight: '#F94D2A',
    redDark: '#985A5A',
  },
  shadows: {
    level1: '0 0 3px rgba(0, 0, 0, 0.03)',
    level2: '0 0 3px rgba(0, 0, 0, 0.3)',
    level3: '0 0 5px rgba(0, 0, 0, 0.3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Noto Sans, sans-serif',
    text: 'Nunito, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
    level5: 100000,
  },
  fontSizes: {
    level1: '1.3rem',
    level2: '1.4rem',
    level3: '1.5rem',
    level4: '1.7rem',
    level5: '2.2rem',
    level6: '2.6rem',
  },
  durations: {
    level1: 0.1,
    level2: 0.2,
    level3: 0.3,
  },
  lineHeights: {
    level1: '1',
    level2: '1.1',
    level3: '1.2',
    level4: '1.3',
    level5: '1.4',
    level6: '1.5',
  },
};

export type ThemeSpacing = keyof typeof theme.spacings;
export type ThemeZIndex = keyof typeof theme.zIndexes;
export type ThemeFontSize = keyof typeof theme.fontSizes;
export type ThemeDuration = keyof typeof theme.durations;
export type ThemeLineHeight = keyof typeof theme.lineHeights;

export default theme;
