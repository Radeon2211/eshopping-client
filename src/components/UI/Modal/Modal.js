import React, { useCallback } from 'react';
import { AnimatePresence } from 'framer-motion';
import { useSelector, useDispatch } from 'react-redux';
import * as actions from '../../../store/actions/indexActions';
import * as modalTypes from '../../../store/actions/modalTypes';
import * as SC from './Modal.sc';
import MyIcon from '../MyIcon/MyIcon';
import LoadingOverlay from '../LoadingOverlay/LoadingOverlay';
import { ReactComponent as PlusIcon } from '../../../images/SVG/plus.svg';
import { backdropVariants, modalVariants } from '../../../shared/framer';
import Signup from '../../../containers/Forms/Signup/Signup';
import Login from '../../../containers/Forms/Login';
import AddProduct from '../../../containers/Forms/AddProduct/AddProduct';

const Modal = () => {
  const isFormLoading = useSelector((state) => state.ui.isFormLoading);
  const isModalOpen = useSelector((state) => state.ui.isModalOpen);
  const modalContent = useSelector((state) => state.ui.modalContent);

  const dispatch = useDispatch();
  const onSetModal = useCallback((isOpen, content) => dispatch(actions.setModal(isOpen, content)), [
    dispatch,
  ]);

  const loadingOverlay = isFormLoading ? (
    <LoadingOverlay alignLoader="center" loaderSize="small" />
  ) : null;

  let modalContentNode = null;
  switch (modalContent) {
    case modalTypes.SIGNUP:
      modalContentNode = <Signup />;
      break;
    case modalTypes.LOGIN:
      modalContentNode = <Login />;
      break;
    case modalTypes.ADD_PRODUCT:
      modalContentNode = <AddProduct />;
      break;
    default:
      break;
  }

  return (
    <AnimatePresence exitBeforeEnter>
      {isModalOpen && (
        <SC.Wrapper>
          <SC.Backdrop
            variants={backdropVariants}
            initial="hidden"
            animate="visible"
            exit="hidden"
            onClick={() => onSetModal(false, '')}
            onKeyDown={() => onSetModal(false, '')}
            tabIndex="0"
            role="button"
            aria-label="Close modal"
          />
          <SC.Popup variants={modalVariants} initial="hidden" animate="visible" exit="hidden">
            <MyIcon size="medium" onClick={() => onSetModal(false, '')} className="close-icon">
              <PlusIcon />
            </MyIcon>
            {loadingOverlay}
            {modalContentNode}
          </SC.Popup>
        </SC.Wrapper>
      )}
    </AnimatePresence>
  );
};

export default Modal;
