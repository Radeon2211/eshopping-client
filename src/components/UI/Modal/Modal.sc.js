import styled from 'styled-components';
import { motion } from 'framer-motion';

export const Wrapper = styled(motion.div)`
  align-items: center;
  display: flex;
  height: 100vh;
  justify-content: center;
  left: 0;
  padding: ${({ theme }) => theme.spacings.level2};
  position: fixed;
  top: 0;
  width: 100%;
  z-index: ${({ theme }) => theme.zIndexes.level4};
`;

export const Backdrop = styled(motion.div)`
  align-items: center;
  background-color: rgba(0, 0, 0, 0.35);
  display: flex;
  height: 100vh;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: ${({ theme }) => theme.zIndexes.level1};
`;

export const Popup = styled(motion.div)`
  background-color: #fff;
  border-radius: 1px;
  max-width: 100%;
  padding: ${({ theme }) => theme.spacings.level3};
  position: relative;
  text-align: center;
  width: 55rem;
  z-index: ${({ theme }) => theme.zIndexes.level2};

  & .close-icon {
    cursor: pointer;
    display: flex;
    position: absolute;
    top: ${({ theme }) => theme.spacings.level1};
    transform: rotate(45deg);
    right: ${({ theme }) => theme.spacings.level1};
  }
`;
