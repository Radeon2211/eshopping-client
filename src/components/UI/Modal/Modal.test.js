import React from 'react';
import { mount } from 'enzyme';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { Router } from 'react-router-dom';
import Modal from './Modal';
import theme from '../../../styled/theme';
import { modalTypes } from '../../../shared/constants';
import {
  defaultUserProfile,
  defaultDeliveryAddress,
} from '../../../shared/testUtility/testUtility';
import * as actions from '../../../store/actions/indexActions';

const mockStore = configureMockStore([thunk]);

const defaultHistory = {
  listen: jest.fn(),
  location: { pathname: '' },
};

afterEach(cleanup);

const setUpEnzyme = (isModalOpen = false, modalContent = '', isFormLoading = false) => {
  const store = mockStore({
    ui: { isModalOpen, modalContent, isFormLoading },
    auth: { profile: defaultUserProfile, deliveryAddress: defaultDeliveryAddress, cart: [] },
    product: { productDetails: {} },
  });

  return mount(
    <Provider store={store}>
      <Router history={defaultHistory}>
        <ThemeProvider theme={theme}>
          <Modal />
        </ThemeProvider>
      </Router>
    </Provider>,
  );
};

const setUpRTL = (isModalOpen = false, modalContent = '', isFormLoading = false) => {
  const store = mockStore({
    ui: { isModalOpen, modalContent, isFormLoading },
    auth: { profile: defaultUserProfile, deliveryAddress: defaultDeliveryAddress, cart: [] },
    product: { productDetails: {} },
  });
  store.dispatch = jest.fn();

  return {
    ...render(
      <Provider store={store}>
        <Router history={defaultHistory}>
          <ThemeProvider theme={theme}>
            <Modal />
          </ThemeProvider>
        </Router>
      </Provider>,
    ),
    store,
  };
};

describe('<Modal />', () => {
  describe('Check how renders', () => {
    it('should render everything correctly', () => {
      const { asFragment } = setUpRTL(true, modalTypes.ABOUT_WEBSITE, false);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should NOT render anything if isModalOpen is false', () => {
      const { asFragment } = setUpRTL(false);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render <LoadingOverlay /> if isFormLoading true', () => {
      const wrapper = setUpEnzyme(true, modalTypes.ABOUT_WEBSITE, true);
      expect(wrapper.find('LoadingOverlay')).toHaveLength(1);
    });

    it('should render <AddProduct />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.ADD_PRODUCT);
      expect(wrapper.find('AddProduct')).toHaveLength(1);
    });

    it('should render <AddAdmin />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.ADD_ADMIN);
      expect(wrapper.find('AddAdmin')).toHaveLength(1);
    });

    it('should render <AboutWebsite />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.ABOUT_WEBSITE);
      expect(wrapper.find('AboutWebsite')).toHaveLength(1);
    });

    it('should render <BuyProducts />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.BUY_PRODUCTS);
      expect(wrapper.find('BuyProducts')).toHaveLength(1);
    });

    it('should render <CartItemAdded />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CART_ITEM_ADDED);
      expect(wrapper.find('CartItemAdded')).toHaveLength(1);
    });

    it('should render <BuyProducts />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.BUY_PRODUCTS);
      expect(wrapper.find('BuyProducts')).toHaveLength(1);
    });

    it('should render <CartItemAdded />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CART_ITEM_ADDED);
      expect(wrapper.find('CartItemAdded')).toHaveLength(1);
    });

    it('should render <ChangeName />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CHANGE_NAME);
      expect(wrapper.find('ChangeName')).toHaveLength(1);
    });

    it('should render <ChangeEmail />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CHANGE_EMAIL);
      expect(wrapper.find('ChangeEmail')).toHaveLength(1);
    });

    it('should render <ChangePhoneNumber />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CHANGE_PHONE_NUMBER);
      expect(wrapper.find('ChangePhoneNumber')).toHaveLength(1);
    });

    it('should render <ChangeAddress />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CHANGE_ADDRESS);
      expect(wrapper.find('ChangeAddress')).toHaveLength(1);
    });

    it('should render <ChangeContacts />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CHANGE_CONTACTS);
      expect(wrapper.find('ChangeContacts')).toHaveLength(1);
    });

    it('should render <ChangePassword />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CHANGE_PASSWORD);
      expect(wrapper.find('ChangePassword')).toHaveLength(1);
    });

    it('should render <ChangeDeliveryAddress />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CHANGE_DELIVERY_ADDRESS);
      expect(wrapper.find('ChangeDeliveryAddress')).toHaveLength(1);
    });

    it('should render <ClearCart />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.CLEAR_CART);
      expect(wrapper.find('ClearCart')).toHaveLength(1);
    });

    it('should render <DeleteAccount />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.DELETE_ACCOUNT);
      expect(wrapper.find('DeleteAccount')).toHaveLength(1);
    });

    it('should render <DeleteProduct />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.DELETE_PRODUCT);
      expect(wrapper.find('DeleteProduct')).toHaveLength(1);
    });

    it('should render <EditProduct />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.EDIT_PRODUCT);
      expect(wrapper.find('EditProduct')).toHaveLength(1);
    });

    it('should render <Login />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.LOGIN);
      expect(wrapper.find('Login')).toHaveLength(1);
    });

    it('should render <RemoveAdmin />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.REMOVE_ADMIN);
      expect(wrapper.find('RemoveAdmin')).toHaveLength(1);
    });

    it('should render <ResetPassword />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.RESET_PASSWORD);
      expect(wrapper.find('ResetPassword')).toHaveLength(1);
    });

    it('should render <PendingUserInfo />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.PENDING_USER_INFO);
      expect(wrapper.find('PendingUserInfo')).toHaveLength(1);
    });

    it('should render <Signup />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.SIGNUP);
      expect(wrapper.find('Signup')).toHaveLength(1);
    });

    it('should render <SendVerificationLink />', () => {
      const wrapper = setUpEnzyme(true, modalTypes.SEND_VERIFICATION_LINK);
      expect(
        wrapper.text().includes('Are you sure? Number of emails to be sent is very limited'),
      ).toEqual(true);
    });
  });

  describe('Check redux actions calls', () => {
    it('should call setModal() after clicking at backdrop', () => {
      const { store } = setUpRTL(true, modalTypes.ABOUT_WEBSITE);
      expect(store.dispatch).not.toHaveBeenCalled();
      fireEvent.click(screen.getByTestId('Modal-backdrop'));
      expect(store.dispatch).toHaveBeenCalledWith(actions.setModal(false));
    });

    it('should call setModal() after clicking at close icon', () => {
      const { store } = setUpRTL(true, modalTypes.ABOUT_WEBSITE);
      expect(store.dispatch).not.toHaveBeenCalled();
      fireEvent.click(screen.getByTestId('Modal-backdrop'));
      expect(store.dispatch).toHaveBeenCalledWith(actions.setModal(false));
    });
  });
});
