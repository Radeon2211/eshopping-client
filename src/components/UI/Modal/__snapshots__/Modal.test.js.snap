// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> check how renders should NOT render anything if modalContent is empty 1`] = `<DocumentFragment />`;

exports[`<Modal /> check how renders should render everything correctly 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  padding: 1.2rem;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 100000;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(0,0,0,0.35);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 10;
}

.c2 {
  background-color: #fff;
  border-radius: 1px;
  max-width: 100%;
  padding: 2.4rem;
  position: relative;
  text-align: center;
  width: 55rem;
  z-index: 100;
}

.c2 .close-icon {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 0.8rem;
  right: 0.8rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 > svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  fill: #777;
  width: 2.5rem;
  height: 2.5rem;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c5 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  line-height: 1.4;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  text-align: justify;
  font-size: 1.5rem;
}

.c4 {
  font-size: 2.4rem;
  font-family: Noto Sans,sans-serif;
  line-height: 1;
  margin-bottom: 2.4rem;
  margin-top: 0;
  text-align: center;
  margin-bottom: 2.4rem;
}

<div
    class="c0"
    data-testid="Modal"
  >
    <div
      aria-label="Close modal"
      class="c1"
      data-testid="Modal-backdrop"
      role="button"
      style="opacity: 0; pointer-events: none;"
      tabindex="0"
    />
    <div
      class="c2"
      data-testid="Modal-ABOUT_WEBSITE"
      style="opacity: 0; pointer-events: none;"
    >
      <div
        class="c3 close-icon"
        color=""
        data-testid="Modal-close-icon"
      >
        <svg>
          plus.svg
        </svg>
      </div>
      <h3
        class="c4"
      >
        About website
      </h3>
      <span
        class="c5"
        color=""
        display="inline-block"
        overflow="visible"
        size="3"
        spacing="0px"
        transform=""
      >
        This website is not used to buy and sell products. Offers on this website are not real and orders will not be realized. Data that you specify in forms do not need to be real. However they are not processed in any way except of storing in database and displaying them on this website.
      </span>
    </div>
  </div>
</DocumentFragment>
`;
